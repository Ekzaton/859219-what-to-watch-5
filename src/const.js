export const APIRoute = {
  FILMS: `/films/`,
  FAVORITE: `/favorite/`,
  PROMO: `/films/promo/`,
  COMMENTS: `/comments/`,
  LOGIN: `/login/`
};

export const AppRoute = {
  ROOT: `/`,
  SIGN_IN: `/login/`,
  MY_LIST: `/mylist/`,
  REVIEW: `/review/`,
  REVIEW_ID: `/films/:id/review/`,
  PLAYER: `/player/`,
  PLAYER_ID: `/player/:id/`,
  FILMS: `/films/`,
  FILMS_ID: `/films/:id/`
};

export const Tab = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const RatingLevel = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
  UNRATED: `Unrated`
};

export const Review = {
  MIN_LENGTH: 50,
  MAX_LENGTH: 400,
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const SignIn = {
  EMAIL: `email`,
  PASSWORD: `password`,
};

export const DEFAULT_GENRE = `All genres`;

export const DEFAULT_MOVIES_COUNT = 8;

export const REG_EXP_EMAIL = /^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/;

export const REG_EXP_PASSWORD = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
