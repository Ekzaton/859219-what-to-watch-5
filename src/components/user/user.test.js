import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {Route, BrowserRouter} from "react-router-dom";

import {AuthorizationStatus} from "../../const";

import User from "./user";

describe(`User snapshot testing`, () => {
  it(`renders component correctly (without Auth)`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_USER: {
        status: AuthorizationStatus.NO_AUTH,
      },
    });

    const userComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <User/>
            </Route>
          </BrowserRouter>
        </Provider>
    );

    expect(userComponent.toJSON()).toMatchSnapshot();
  });

  it(`renders component correctly (with Auth)`, () => {
    const mockStore = configureStore([]);

    const store = mockStore({
      APP_USER: {
        status: AuthorizationStatus.AUTH,
      },
    });

    const userComponent = renderer.create(
        <Provider store={store}>
          <BrowserRouter>
            <Route>
              <User/>
            </Route>
          </BrowserRouter>
        </Provider>
    );

    expect(userComponent.toJSON()).toMatchSnapshot();
  });
});
