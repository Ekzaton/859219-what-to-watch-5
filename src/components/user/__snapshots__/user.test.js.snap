// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User snapshot testing renders component correctly (with Auth) 1`] = `
<div
  className="user-block__avatar"
>
  <a
    href="/mylist/"
    onClick={[Function]}
  >
    <img
      alt="User avatar"
      height="63"
      src="/img/avatar.jpg"
      width="63"
    />
  </a>
</div>
`;

exports[`User snapshot testing renders component correctly (without Auth) 1`] = `
<div
  className="user-block__link"
>
  <a
    className="user-block__link"
    href="/login/"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
